### DISCUSSION TITLE
Validate Array Field

### DISCUSSION THREAD

**Discussion by Joan-George** [react-hook-form/react-hook-form#9828] [Q&A]
How can I validate duplicate values in the array fields?

**Comment by Moshyfawn** [comment]
Depending on the validation strategy you use, with "native" `useFieldArray` validation rules, you can do:
```js
useFieldArray({
  rules: {
    validate: arr => checkDuplicateValues(arr)
  }
})
```
To get the error
```js
errors?.fieldArray?.root?.message
```

**Reply by Joan-George** [reply]
Is checkDuplicateValues is our own custom function?

If you could give me some examples that would be helpful.

**Reply by Moshyfawn** [reply]
`checkDuplicateValues` is whatever logic you want to run to validate your array fields.

**Comment by Sajarin-M** [comment]
Yes, `checkDuplicateValues` would be a custom function that you would need to write yourself.
Here is an example of `checkDuplicateValues` function for an array of strings:
```js
function checkDuplicateValues(arr) {
  // Create a set from the array
  let set = new Set(arr);
  if (set.size !== arr.length) {
    return "duplicate values found";
  }
}
```

**Reply by Moshyfawn** [reply]
Thank you, Sajarin!
